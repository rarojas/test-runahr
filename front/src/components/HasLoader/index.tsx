import React from 'react';
import Loader from '../Loader';
import styled from 'styled-components';

const Wrapper = styled.div`
  position: relative;
`;

const Overlay = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(0.3, 0.3, 0.3, 0.3);
  justify-content: center;
  align-items: center;
  display: flex;
`;

export interface HasLoaderProps {
  loading?: boolean;
}

const HasLoader = <P extends object>(Component: React.ComponentType<P>) => (
  props: HasLoaderProps & P
) => (
  <Wrapper>
    <Component {...props} />
    {props.loading && (
      <Overlay>
        <Loader />
      </Overlay>
    )}
  </Wrapper>
);

HasLoader.defaultProps = {
  loading: false
};

export default HasLoader;
