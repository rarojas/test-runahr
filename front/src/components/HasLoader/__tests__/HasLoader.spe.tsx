import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import HasLoader from '../index';
import {
  Button
  //@ts-ignore
} from 'styled-bootstrap-components';

const TestHasLoader = HasLoader(Button);

const defaultProps = {
    loading: false
  },
  EnchancedComponent = (props: any) => (
    <TestHasLoader {...defaultProps} {...props} />
  );

describe('Render DateInput', () => {
  it('render correctly date component', () => {
    const DateInputComponent = renderer.create(<EnchancedComponent />).toJSON();
    expect(DateInputComponent).toMatchSnapshot();
  });

  it('render date input correctly with empty value', () => {
    const props = {
        loading: true
      },
      DateInputComponent = mount(<EnchancedComponent {...props} />);
    expect(DateInputComponent.prop('loading')).toEqual(true);
  });
});
