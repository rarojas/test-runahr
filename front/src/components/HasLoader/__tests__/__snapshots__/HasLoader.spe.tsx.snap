// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DateInput render correctly date component 1`] = `
<div
  className="sc-bwzfXH ghuPdm"
>
  <button
    className="sc-gzVnrw eUsuSq"
  />
</div>
`;
